pkg/util/http/transport.go

```diff
@@ -57,46 +68,75 @@ func CreateHTTPTransport() *http.Transport {
 // GetProxyTransportFunc return a proxy function for a http.Transport that
 // would return the right proxy depending on the configuration.
 func GetProxyTransportFunc(p *config.Proxy) func(*http.Request) (*url.URL, error) {
```

Comment:
Could we extend the existing test to test nonexect proxy matches (not necessary the `httpproxy` package but the fact it's correctly used).